crates/fuel-networks/src/lib.rs

```diff
@@ -0,0 +1,75 @@
+/// FuelStreamsNetworks; shortened to FuelNetworks for brievity and public familiarity
+
+#[derive(Debug, Clone, Default)]
+pub enum FuelNetworkUserRole {
+    Admin,
+    #[default]
+    Default,
+}
+
+#[derive(Debug, Copy, Clone, Default, clap::ValueEnum)]
+pub enum FuelNetwork {
+    #[default]
+    Local,
+    Testnet,
+    Mainnet,
+}
+
+impl std::fmt::Display for FuelNetwork {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            FuelNetwork::Local => write!(f, "local"),
+            FuelNetwork::Testnet => write!(f, "testnet"),
+            FuelNetwork::Mainnet => write!(f, "mainnet"),
+        }
+    }
+}
+
+impl FuelNetwork {
+    pub fn load_from_env() -> Self {
+        match std::env::var("NETWORK").as_deref() {
+            Ok("testnet") => FuelNetwork::Testnet,
+            Ok("mainnet") => FuelNetwork::Mainnet,
+            _ => FuelNetwork::Local,
+        }
+    }
+
+    pub fn to_nats_url(&self) -> String {
```

Comment:
By pinning the NATS URL here, we will have the same issue we had in the past: not being able to pass a custom `NATS_URL` to internal cluster traffic. As mentioned previously, when running this internally in the cluster, instead of going directly to the NATS node host `nats://nats.<namespace>.svc.cluster.local:4222`, we will always go through the external service NLB.

Reply:
Please see: https://github.com/FuelLabs/data-systems/blob/d554ed1474208015a666173cfd849bc28bd30308/crates/fuel-streams-storage/src/nats/nats_client_opts.rs#L70,

That's where we use the env variable for the admin role. For default roles, we will be using these hardcoded values